### PR TITLE
treewide: use new tmpfiles API

### PR DESCRIPTION
**Pull request by h7x4**
## Description of changes

This PR revamps some of the `tmpfiles` configurations in a few modules to use the new API provided by #262731. While the change does not contribute anything new in terms of functionality, it is meant to kickstart the usage of the API among new PRs, where the PR author might look to other modules for reference. These are supposed to serve as such references.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — COMMENTED** [review]
LGTM, but I would wait for another review.

**Review by pbsds — APPROVED** [review]

File: nixos/modules/services/networking/charybdis.nix
Comment (ambroisie) [inline nixos/modules/services/networking/charybdis.nix:84]: Missing `10-charybdis`?

```suggestion
      systemd.tmpfiles.settings."10-charybdis".${cfg.statedir}.d = {
```

**Review by ambroisie — CHANGES_REQUESTED** [review]
One nit, otherwise LGTM.

And a question: are we settling on `10-<service-name>` as the default name to use for the settings file? I'm not sure what the split is currently, nixpkgs-wide (between numbered prefix and bare service name), but that seems to go _somewhat_ against the spirit of just using the NixOS module system IMO (since it implies that overriding the _files_ through lexical order is expected).

Anyway, I haven't really seen anybody argue for one or the other yet: I'm just curious if there was some conversation about that point I might have missed.

File: nixos/modules/services/networking/charybdis.nix
Comment (h7x4) [reply nixos/modules/services/networking/charybdis.nix:84]: Yup, that's a slip :)

**Comment by h7x4** [comment]
> Are we settling on 10-<service-name> as the default name to use for the settings file? I'm not sure what the split is currently, nixpkgs-wide (between numbered prefix and bare service name), but that seems to go somewhat against the spirit of just using the NixOS module system IMO (since it implies that overriding the files through lexical order is expected).
>
> Anyway, I haven't really seen anybody argue for one or the other yet: I'm just curious if there was some conversation about that point I might have missed.

Good question!

I chose to blindly follow the format that was used in the example added in original PR for arbitrary reasons. But I have seen other people use it without the number prefix. The only discussion I can find related to this topic would be https://github.com/NixOS/nixpkgs/pull/262731#discussion_r1367894061, which seems to come to the conclusion that it's best to leave it up to the module author and user to choose. Maybe we could use this PR for extended discussion?

AFAIK, the settings inside these kinds of files aren't really ordered in any meaningful way, so adding new rules for the same service into a different file doesn't make sense to me. And if you'd need to override something like the user or the group, it could easily be done through the module system, potentially requiring use of `mkForce`. Maybe there's a case to be made for packages that vendor their own tmpfiles configuration, if any?

Do you have any thoughts @arianvp, @Lassulus, @aanderse, @nikstur?

**Review by ambroisie — APPROVED** [review]
LGTM, barring the filename discussion.

**Comment by nikstur** [comment]
@SuperSandro2000 @Janik-Haag sorry I only meant to request Lassulus review, but GitHub mobile decided that your review is also required :D

**Comment by Lassulus** [comment]
The 10- prefix looks fine to me. Not that it matters a lot but seems like a good default.
Entries are sorted alphabetically so parent directories are listed in there before their children. That was fine in our tests and we haven't encountered any issue yet.

I'm a bit unsure if we handle spaces in directories correctly though. I have to test that later today.

**Review by Lassulus — APPROVED** [review]

**Pull request merged**
